mfbt/Assertions.h

```diff
@@ -50,7 +50,21 @@ MOZ_BEGIN_EXTERN_C
 extern MFBT_DATA const char* gMozCrashReason;
 MOZ_END_EXTERN_C
 
-#if defined(MOZ_HAS_MOZGLUE) || defined(MOZILLA_INTERNAL_API)
+#if defined(__HAIKU__)
+/*
+ * We introduce a function prototype here as including Haiku specific header
+ * from this globally included header will cause too many conflicts.
+ */
+MOZ_BEGIN_EXTERN_C
+// Workaround for gcc_hidden.h hack of libxul.so
+#pragma GCC visibility push(default)
+extern void debugger(const char *message);
+#pragma GCC visibility pop
+MOZ_END_EXTERN_C
+
+#  define MOZ_CRASH_ANNOTATE(...) debugger(__VA_ARGS__)
```

Comment:
`debugger` unfortunately only takes one `const char*`. So if there are format arguments passed here we have to handle that a different way. Though looking at the next method, it looks like there's only one argument to `AnnotateMozCrashReason`, so maybe this will work anyway?

Reply:
I wonder that, but yes this built successfully.